test/k8s_service_example_test_helpers.go

```diff
@@ -115,3 +183,66 @@ func verifyServiceAvailable(
 		validationFunction,
 	)
 }
+
+// verifyServiceRoutesToMainAndCanaryPods ensures that the service is routing to both the main and the canary pods
+// It does this by repeatedly issuing requests to the service and inspecting the nginx Server header
+// Once both nginx tags have been seen in this header - we can be confident that we've reached both types of pod via the service
+func verifyServiceRoutesToMainAndCanaryPods(
+	t *testing.T,
+	kubectlOptions *k8s.KubectlOptions,
+	appName string,
+	releaseName string,
+) {
+
+	// Get the service and wait until it is available
+	filters := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/instance=%s", appName, releaseName),
+	}
+	services := k8s.ListServices(t, kubectlOptions, filters)
+	require.Equal(t, len(services), 1)
+	service := services[0]
+	k8s.WaitUntilServiceAvailable(t, kubectlOptions, service.Name, WaitTimerRetries, WaitTimerSleep)
+
+	var availableService = *k8s.GetService(t, kubectlOptions, service.Name)
+	serviceEndpoint := k8s.GetServiceEndpoint(t, kubectlOptions, &availableService, 80)
+
+	// Ensure that the service routes to both the main and canary deployment pods
+	// Read the latest values dynamically in case the fixtures file changes
+	valuesFile, err := ioutil.ReadFile(filepath.Join("fixtures", "canary_and_main_deployment_values.yaml"))
+	assert.NoError(t, err)
+
+	rendered := map[string]interface{}{}
+	err = yaml.Unmarshal([]byte(valuesFile), &rendered)
+
+	mainImageTag := rendered["containerImage"].(map[string]interface{})["tag"].(string)
+
+	canaryImageTag := rendered["canary"].(map[string]interface{})["containerImage"].(map[string]interface{})["tag"].(string)
+
+	// We haven't seen either tag come back in the nginx Server header yet
+	seen := make(map[string]bool)
+	seen[mainImageTag] = false
+	seen[canaryImageTag] = false
+
+	maxRetries := 30
+	sleepDuration := 1 * time.Second
```

Comment:
NIT: Add a comment that indicates how long this will take to timeout (30 seconds: 30 max tries * 1 second between each try).

Reply:
:+1:  Added in edb0d90